### PR TITLE
chore(hlapi): add InnerCiphertext type to integer wrapper

### PR DESCRIPTION
**Pull request by tmontaigu**
Make the GenericInteger struct have a generic `InnerCiphertext` instead of always RadixCiphertext.

This is to prepare the addition of signed types which will use a SignedRadixCiphertext.

### CODE REVIEW

File: tfhe/src/high_level_api/integers/keys.rs [hunk 1]
```diff
@@ -196,22 +196,6 @@ impl IntegerCompactPublicKey {
         Some(Self { key: Some(key) })
     }
 
-    pub(in crate::high_level_api) fn try_encrypt<T>(
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/keys.rs:199]: why is this removed ?

File: tfhe/src/high_level_api/integers/types/base.rs [hunk 2]
```diff
@@ -932,25 +1055,25 @@ generic_integer_impl_operation!(Mul(mul) => mul_parallelized);
...
-generic_integer_impl_operation!(RotateLeft(rotate_left) => rotate_left_parallelized);
-generic_integer_impl_operation!(RotateRight(rotate_right) => rotate_right_parallelized);
 generic_integer_impl_operation!(Div(div) => div_parallelized);
 generic_integer_impl_operation!(Rem(rem) => rem_parallelized);
-
+generic_integer_impl_shift_rotate!(Shl(shl) => left_shift_parallelized);
+generic_integer_impl_shift_rotate!(Shr(shr) => right_shift_parallelized);
+generic_integer_impl_shift_rotate!(RotateLeft(rotate_left) => rotate_left_parallelized);
+generic_integer_impl_shift_rotate!(RotateRight(rotate_right) => rotate_right_parallelized);
+//
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/types/base.rs:1064]: comment what for ?

File: tfhe/src/high_level_api/integers/keys.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/keys.rs:199]: Because its not needed anymore

File: tfhe/src/high_level_api/integers/types/base.rs (hunk 2 above)
Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/types/base.rs:1064]: It was meant to have a comment like `assign operations` to space out macro calls

File: tfhe/src/high_level_api/integers/keys.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/keys.rs:199]: That thing is now handled line 51 of [tfhe/src/high_level_api/integers/client_key.rs](https://github.com/zama-ai/tfhe-rs/pull/601/files#diff-db1fc01f72b02f7ba047502431176a57f4df20496246ad61e1e35c01fb7a69ab)

Comment (IceTDrinker) [reply tfhe/src/high_level_api/integers/keys.rs:199]: ok thanks !

**Review by IceTDrinker — APPROVED** [review]
Thanks !
